src/main/java/org/testng/ITestListener.java

```diff
@@ -60,6 +60,15 @@ default void onTestFailedButWithinSuccessPercentage(ITestResult result) {
     // not implemented
   }
 
+  /**
+   * Invoked each time a test fails due to a timeout.
+   *
+   * @param result <code>ITestResult</code> containing information about the run test
+   */
+  default void onTestFailedWithTimeout(ITestResult result) {
+    // not implemented
```

Comment:
Default should call `onTestFailure`

Reply:
Can you please let me know why so? We are anyways invoking `onTestFailedWithTimeout()` followed by `onTestFailure()` explicitly in `TestListenerHelper`. Then why do we need to chain things here?